stablehlo/reference/Ops.cpp

```diff
@@ -500,6 +521,27 @@ SmallVector<InterpreterValue> eval(
       auto result = evalReducePrecisionOp(operand, exponentBits, mantissaBits,
                                           reducePrecisionOp.getType());
       scope.add(reducePrecisionOp.getResult(), result);
+    } else if (auto reduceScatterOp = dyn_cast<ReduceScatterOp>(op)) {
+      auto operand = scope.findTensor(reduceScatterOp.getOperand());
+      int64_t scatterDimension = reduceScatterOp.getScatterDimension();
+      auto replicaGroupsAttr = reduceScatterOp.getReplicaGroups();
+      auto replicaGroupsShape = replicaGroupsAttr.getShapedType().getShape();
+      SmallVector<SmallVector<uint32_t>> replicaGroups(replicaGroupsShape[0]);
+      auto replicaGroupsIt = replicaGroupsAttr.getValues<int64_t>().begin();
+      for (auto &replicaGroup : replicaGroups)
+        for (auto i = 0; i < replicaGroupsShape[1]; ++i, ++replicaGroupsIt)
+          replicaGroup.push_back(*replicaGroupsIt);
```

Comment:
Should probably factor the conversion from getReplicaGroups to SmallVector<SmallVector<uint32_t>> into a helper function - this is non-trivial duplication of non-trivial logic.